# HG changeset patch
# User Philip Nienhuis <prnienhuis@users.sf.net>
# Date 1471618879 -7200
#      Fri Aug 19 17:01:19 2016 +0200
# Node ID 929a1500907ee9b1cb17ae4bca0d4af771efa74c
# Parent  dc73bbd66363ae7054e76964441ee21ee51171bc
uigetfile.m: fix type and allow path names as input arg (bug #48828)
* uigetfile.m: add stanza about path name / directory name in texinfo header
* private/__file_filter__.m: fix typo; check for empty retval before indexing it

diff --git a/scripts/gui/private/__file_filter__.m b/scripts/gui/private/__file_filter__.m
--- a/scripts/gui/private/__file_filter__.m
+++ b/scripts/gui/private/__file_filter__.m
@@ -1,9 +1,9 @@
-## Copyright (C) 2010-2016 Kai Habel
+## Copyright (C) 2010-2015 Kai Habel
 ##
 ## This file is part of Octave.
 ##
 ## Octave is free software; you can redistribute it and/or modify it
 ## under the terms of the GNU General Public License as published by
 ## the Free Software Foundation; either version 3 of the License, or (at
 ## your option) any later version.
 ##
@@ -21,17 +21,17 @@
 ## Undocumented internal function.
 ## @end deftypefn
 
 ## Author: Kai Habel
 
 function [retval, defname, defdir] = __file_filter__ (caller, file_filter)
 
   #keyboard;
-  revtal = {};
+  retval = {};
   defname = "";
   defdir = "";
 
   if (nargin == 1 || isempty (file_filter))
     ## Do nothing, and just add default pattern.
   elseif (iscell (file_filter))
     [r, c] = size (file_filter);
     if (c != 1 && c != 2)
@@ -58,17 +58,19 @@ function [retval, defname, defdir] = __f
     endif
     if (! isempty (fext))
       fext = ["*" fext];
       retval = {fext, __default_filtername__(fext)};
     endif
   endif
 
   ## Delete any "*.*" pattern, and add "* All Files"
-  retval(strcmp (retval(1,:), "*.*"), :) = [];
+  if (! isempty (retval))
+    retval(strcmp (retval(1,:), "*.*"), :) = [];
+  endif
   retval(end+1,:) = {"*", __default_filtername__("*")};
 
 endfunction
 
 function name = __default_filtername__ (filterext)
 
   name = "";
 
diff --git a/scripts/gui/uigetfile.m b/scripts/gui/uigetfile.m
--- a/scripts/gui/uigetfile.m
+++ b/scripts/gui/uigetfile.m
@@ -44,16 +44,26 @@
 ## the second column.
 ## Example: @code{uigetfile (@{"*.ext", "My Description";"*.xyz",
 ## "XYZ-Format"@})}
 ##
 ## The filter string can also contain a semicolon separated list of filter
 ## extensions.
 ## Example: @code{uigetfile (@{"*.gif;*.png;*.jpg", "Supported Picture
 ## Formats"@})}
+##
+## @item A directory name or path name
+## If the folder name of path name contains a trailing file separator, the
+## contents of that folder will be displayed.  If no trailing file separator
+## is present the parent directory is listed.  The substring to the right of
+## the rightmost file separator (if any) will be interpreted as a file or
+## directory name and if that file or directory exists it will be highlighted.
+## If the path name or directory name is wholly or partly nonexistent, the
+## current working directory will be displayed.
+## No filter will be active.
 ## @end table
 ##
 ## @var{dialog_name} can be used to customize the dialog title.
 ##
 ## If @var{default_file} is given then it will be selected in the GUI dialog.
 ## If, in addition, a path is given it is also used as current path.
 ##
 ## The screen position of the GUI dialog can be set using the
